packages/cdktf/lib/terraform-resource.ts

```diff
@@ -50,6 +50,32 @@ export interface TerraformResourceLifecycle {
   readonly postcondition?: Postcondition[];
 }
 
+/**
+ * prepares a lifecycle object for being rendered as JSON
+ * currently this function:
+ *  - converts all replaceTriggeredBy items that are ITerraformDependables to strings
+ */
+export function lifecycleToTerraform(
+  lifecycle?: TerraformResourceLifecycle
+): TerraformResourceLifecycle | undefined {
+  if (!lifecycle) {
+    return undefined;
+  }
+
+  return {
+    ...lifecycle,
+    ...(lifecycle?.replaceTriggeredBy?.length && {
+      replaceTriggeredBy: lifecycle?.replaceTriggeredBy?.map((x) => {
```

Comment:
Nit: I'd prefer a ternary. The && makes it look brittle to me, I think my frontend instincts are kicking in